scipy/sparse/csgraph/_disjoint_set.pyx

```diff
@@ -0,0 +1,137 @@
+"""
+Disjoint set data structure
+"""
+
+import numpy as np
+cimport numpy as np
+cimport cython
+
+
+cdef class DisjointSet:
+    """ Disjoint set data structure for incremental connectivity queries.
+
+    .. versionadded: 1.6.0
+
+    Parameters
+    ----------
+    n : int
```

Comment:
Perhaps use terminology: `n -> n_nodes`, `nc -> n_components`.

Also, it should be mentioned somewhere that nodes indices are integers from `[0, 1, ..., n_nodes - 1]`.

Reply:
Thanks. I have made these changes.